audit_log/types.py

```diff
@@ -0,0 +1,89 @@
+from dataclasses import dataclass
```

Comment:
Just adding a note here that this data validation might be a good place to use something like `pydantic` or `marshmallow` but it also might be a bit overkill right now. And it's definitely something that can be done later, so maybe no point of investing time on it now.